### PR TITLE
read-json fix request encoding

### PR DESCRIPTION
**Pull request by liuchunhua**

### CODE REVIEW

**Comment by weavejester** [comment]
Thanks for the PR! You seem to have two commits with the same description. Can you fix that? Also can you add a test or two that verifies the change?

File: src/ring/middleware/json.clj [hunk 1]
```diff
@@ -11,7 +12,10 @@
 (defn- read-json [request & [{:keys [keywords? bigdecimals?]}]]
   (if (json-request? request)
     (if-let [body (:body request)]
-      (let [body-string (slurp body)]
+      (let [encoding (or (:character-encoding request)
```
Comment (weavejester) [inline src/ring/middleware/json.clj]: The `:character-encoding` key is deprecated, and should be identical to the output of `character-encoding`. So we can just write:

```clojure
(or (character-encoding request) "UTF-8")
```

**Comment by weavejester** [comment]
Your update somehow slipped through my todo list. If you're still around, could you squash your commits and make sure the first letter is correctly capitalized in the commit message? Thanks.

**Comment by liuchunhua** [comment]
Is that okay?

**Comment by weavejester** [comment]
Looks fine

**Pull request merged**